HKMP/Game/Client/PlayerManager.cs

```diff
@@ -331,26 +355,38 @@ byte skinId
         /// <param name="team">The team that the player is on.</param>
         public void AddNameToPlayer(GameObject playerContainer, string name, Team team = Team.None) {
             // Create a name object to set the username to, slightly above the player object
-            var nameObject = Object.Instantiate(
-                new GameObject("Username"),
-                playerContainer.transform.position + new Vector3(0, 1.25f, 0),
-                Quaternion.identity
-            );
-            nameObject.name = "Username";
-            nameObject.transform.SetParent(playerContainer.transform);
-            nameObject.transform.localScale = new Vector3(0.25f, 0.25f, nameObject.transform.localScale.z);
-            nameObject.AddComponent<KeepWorldScalePositive>();
-
-            // Add a TextMeshPro component to it, so we can render text
-            var textMeshObject = nameObject.AddComponent<TextMeshPro>();
-            textMeshObject.text = name.ToUpper();
-            textMeshObject.alignment = TextAlignmentOptions.Center;
-            textMeshObject.font = FontManager.InGameNameFont;
-            textMeshObject.fontSize = 22;
-            textMeshObject.outlineWidth = 0.2f;
-            textMeshObject.outlineColor = Color.black;
+            var nameObject = playerContainer.FindGameObjectInChildren("Username");
+            TextMeshPro textMeshObject = null;
+
+            if (nameObject == null)
+            {
+                nameObject = new GameObject("Username");
+                nameObject.transform.position = playerContainer.transform.position + Vector3.up * 1.25f;
+                nameObject.name = "Username";
+                nameObject.transform.SetParent(playerContainer.transform);
+                nameObject.transform.localScale = new Vector3(0.25f, 0.25f, nameObject.transform.localScale.z);
+                nameObject.AddComponent<KeepWorldScalePositive>();
+
+                // Add a TextMeshPro component to it, so we can render text
+                textMeshObject = nameObject.AddComponent<TextMeshPro>();
+                textMeshObject.text = "Username";
+                textMeshObject.alignment = TextAlignmentOptions.Center;
+                textMeshObject.font = FontManager.InGameNameFont;
+                textMeshObject.fontSize = 22;
+                textMeshObject.outlineWidth = 0.2f;
+                textMeshObject.outlineColor = Color.black;
+
+                nameObject.transform.SetParent(playerContainer.transform);
+            }
+            else
+            {
```

Comment:
Code-style inconsistency.
```suggestion
            } else {
```